lib/using_restricted_array.rb

```diff
@@ -1,61 +1,117 @@
-require_relative 'restricted_array.rb'
+require_relative "restricted_array.rb"
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear or O(n) time complexity,where n is the number of items in the array or the length of thei nput
+# Space complexity: constant or O(1)
 def length(array)
-  raise NotImplementedError
+  k = 0
+  while array[k] != nil
```

Comment:
What happens if you run over the end of the array? If there's a bug somewhere else, you may end up in a state where the array is not nil-terminated.